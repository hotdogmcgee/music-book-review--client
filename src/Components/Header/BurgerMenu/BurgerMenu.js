import React from "react";
import { slide as Menu } from "react-burger-menu";
import "./BurgerMenu.css";
import TokenService from "../../../services/token-service";
import { Link } from "react-router-dom";

export default class BurgerMenu extends React.Component {
  constructor(props) {
    super(props);

    this.state = {
      menuOpen: false
    };
  }
  handleStateChange (state) {
    this.setState({menuOpen: state.isOpen})  
  }
  
  closeMenu() {
    this.setState({ menuOpen: false });
  }
  handleLogoutClick = () => {
    this.closeMenu();
    TokenService.clearAuthToken();
    TokenService.clearUserId();
    this.props.hasLogin(false);
  };

  handleLoginClick = () => {
    this.closeMenu();
    this.props.showModal();
  };

  renderLoginOrLogout = () => {
    const link = TokenService.hasAuthToken() ? (
      <Link onClick={this.handleLogoutClick} to="/" className="menu-item">
        Logout
      </Link>
    ) : (
      <li onClick={this.handleLoginClick} id="log-in" className="menu-item">
        Log In/Register
      </li>
    );
    return link;
  };

  renderMyProfileLink = () => {
    const profileLink = TokenService.hasAuthToken() ? (<Link onClick={() => this.closeMenu()} className="menu-item" to="/my-profile">
    My Profile
  </Link>) : ""
  return profileLink
  }

  render() {

    
    return (
      <div>
        <Menu isOpen={this.state.menuOpen} onStateChange={(state) => this.handleStateChange(state)} right id="burger-menu">
          {this.renderLoginOrLogout()}
          <Link to="/about" id="about" className="menu-item" onClick={() => this.closeMenu()}>
            About
          </Link>
          {this.renderMyProfileLink()}
        </Menu>
      </div>
    );
  }
}
