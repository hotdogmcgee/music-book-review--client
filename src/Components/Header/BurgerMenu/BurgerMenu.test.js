import React from "react";
import BurgerMenu from "./BurgerMenu";
import { shallow } from "enzyme";
import toJson from "enzyme-to-json";

describe("BurgerMenu component", () => {
  it("renders a BurgerMenu by default", () => {
    const wrapper = shallow(<BurgerMenu />);
    expect(toJson(wrapper)).toMatchSnapshot();
  });
});
