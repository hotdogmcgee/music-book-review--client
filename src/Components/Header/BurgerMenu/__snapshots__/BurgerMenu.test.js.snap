// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`BurgerMenu component renders a BurgerMenu by default 1`] = `
<div>
  <Menu
    bodyClassName=""
    burgerBarClassName=""
    burgerButtonClassName=""
    className=""
    crossButtonClassName=""
    crossClassName=""
    disableAutoFocus={false}
    disableCloseOnEsc={false}
    htmlClassName=""
    id="burger-menu"
    isOpen={false}
    itemClassName=""
    itemListClassName=""
    menuClassName=""
    morphShapeClassName=""
    noOverlay={false}
    noTransition={false}
    onIconHoverChange={[Function]}
    onStateChange={[Function]}
    outerContainerId=""
    overlayClassName=""
    pageWrapId=""
    right={true}
    styles={Object {}}
    width={300}
  >
    <li
      className="menu-item"
      id="log-in"
      onClick={[Function]}
    >
      Log In/Register
    </li>
    <Link
      className="menu-item"
      id="about"
      onClick={[Function]}
      to="/about"
    >
      About
    </Link>
  </Menu>
</div>
`;
